builtin/providers/aws/resource_aws_opsworks_instance_test.go

```diff
@@ -0,0 +1,211 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// These tests assume the existence of predefined Opsworks IAM roles named `aws-opsworks-ec2-role`
+// and `aws-opsworks-service-role`.
+
+func TestAccAWSOpsworksInstance(t *testing.T) {
+	stackName := fmt.Sprintf("tf-%d", acctest.RandInt())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsOpsworksInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAwsOpsworksInstanceConfigCreate(stackName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "hostname", "tf-acc1",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "instance_type", "t2.micro",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "state", "stopped",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "layer_ids.#", "1",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "install_updates_on_boot", "true",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "architecture", "x86_64",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "os", "Amazon Linux 2014.09", // inherited from opsworks_stack_test
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "root_device_type", "ebs", // inherited from opsworks_stack_test
+					),
+					resource.TestCheckResourceAttr(
+						"aws_opsworks_instance.tf-acc", "availability_zone", "us-west-2a", // inherited from opsworks_stack_test
+					),
```

Comment:
After waking up some more and with some more coffee, it makes sense now.  I'm working on adding the tests.